src/module_3/push_exploration.md

```diff
@@ -0,0 +1,5364 @@
+
+From binary: 
+- We would try to use some resampling technique for grouping the 4 into just 1. We will create a column 'any_event' to input value '1' if any of the four cols has a value of 1. In this sense, we address the unbalanced distribution (a bit), we simplify it and also keep the info if any of these events occured
+
+
+
+```python
+# Remove numericals
+numerical_remove = ['count_adults', 'count_children', 'count_babies',
+                    'std_days_to_buy_product_type', 'std_days_to_buy_variant_id']
+numerical = [col for col in numerical if col not in numerical_remove]
+
+# Remove categoricals
+categorical.remove('vendor')
+
+# Create the binary one
+filtered_data['any_event'] = filtered_data[['ordered_before', 'abandoned_before', 'active_snoozed', 'set_as_regular']].any(axis=1).astype(int)
+binary = ['any_event']
+# I'm thinking if maybe would be interesting to perform a sume here. If any_event = 2 would be stronger than 1 and so, while also keeping info about events.
+
+```
+
+    /tmp/ipykernel_481/1254836267.py:10: SettingWithCopyWarning: 
+    A value is trying to be set on a copy of a slice from a DataFrame.
+    Try using .loc[row_indexer,col_indexer] = value instead
+    
+    See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
+      filtered_data['any_event'] = filtered_data[['ordered_before', 'abandoned_before', 'active_snoozed', 'set_as_regular']].any(axis=1).astype(int)
+
+
+Additionally, let's check if order_date is always equal to created_at (maybe some orders are created but not ordered until X days). If so, let's remove order_date (created_at is has also hour and minut info). Maybe hourly data would be crucial for planning the timing of sending notifications.
+
+
+```python
+if len(filtered_data[filtered_data['order_date'] == filtered_data['created_at']]) == len(filtered_data):
+    print("ofc")
+```
+
+
+```python
+information.remove('order_date')
+information
+```
+
+
+
+
+    ['variant_id', 'order_id', 'user_id', 'created_at']
+
+
+
+
+```python
+cols = information + numerical + categorical + binary + predicted
+final_data = filtered_data[cols]
+```
+
+Time format encoding for created_at. Later we could inspect if this granularity it is too much and it is better to leave it at day_of_week info
+
+
+```python
+#Verify created at has pandas date_format
+final_data['created_at'] = pd.to_datetime(final_data['created_at'])
+
+# Extract year, month, day, and hour as separate features
+final_data['year'] = final_data['created_at'].dt.year
+final_data['month'] = final_data['created_at'].dt.month
+# Extract the day of the week (numerical, 1-7, starting with Monday as 1)
+final_data['day_of_week'] = final_data['created_at'].dt.dayofweek + 1
+final_data['hour'] = final_data['created_at'].dt.hour
+final_data = final_data.drop(columns=['created_at'])
+```
+
+    /tmp/ipykernel_481/1071411211.py:2: SettingWithCopyWarning: 
```

Comment:
To avoid this warnings, do `final_data = filtered_data[cols].copy()` when you define `final_data`